OptionSelector/OptionSelectorTableViewController.m

```diff
@@ -0,0 +1,205 @@
+            [defaults setObject:tmp.detailTextLabel.text forKey:@"cell2"];
+        }
+        else{
+            NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+            [defaults setObject:tmp.detailTextLabel.text forKey:@"cell3"];
+        }
+    }
+    
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.array = [[NSMutableArray alloc]init];
+    
+    Type *dogs = [[Type alloc]init];
+    dogs.name = @"Dogs";
+    dogs.options = @[
+                     @"Labrador Retriever",
+                     @"German Shepherd",
+                     @"Golden Retriever ",
+                     @"Bulldog",
+                     @"Yorksire Terrier",
+                     @"Poodle",
+                     @"Beagle",
+                     @"Chihuahua",
+                     @"Dachshund"
+                     ];
+    dogs.selection = @" ";
+    
+    Type *cats = [[Type alloc]init];
+    cats.name = @"Cats";
+    cats.options = @[
+                     @"Cornish Rex",
+                     @"Russian Blue",
+                     @"Balinese Cat",
+                     @"Manx Cat",
+                     @"Abyssinian Cat",
+                     @"Birman",
+                     @"Siberian Cat",
+                     @"Turkish Angora",
+                     @"Norwegian Forest Cat"
+                     ];
+    cats.selection = @" ";
+    
+    Type *cars = [[Type alloc]init];
+    cars.name = @"Cars";
+    cars.options = @[
+                     @"Ferari F-150",
+                     @"Buggati",
+                     @"Mercedes SLR",
+                     @"Doge Dart",
+                     @"Honda Civic",
+                     @"Toyota Corolla",
+                     @"Subaru Impreza",
+                     @"Audi TT",
+                     @"Porsche",
+                     @"Lamborghini"
+                     ];
+    cars.selection = @" ";
+
+    
+    [self.array addObject:dogs];
+    [self.array addObject:cats];
+    [self.array addObject:cars];
+    
+}
+
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.array.count;
+}
+
+-(void)optionSelectorDetailTableViewController:(OptionSelectorDetailTableViewController *)viewController didSelectType:(Type *)type{
+    
+    //This is redundant and unnecessary as the 'Type' object is a pointer and any changes made to it in the detail table view controller will reflect in this table viewController also.
+    [self.array replaceObjectAtIndex:self.index withObject:type];
+    [self.tableView reloadData];
+    
+}
+-(void)viewWillAppear:(BOOL)animated{
+    
+    [super viewWillAppear:animated];
+    [self.tableView reloadData];
+    
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
```

Comment:
See comment https://github.com/accesscode-2-2/unit-1-hw-week-4/pull/13/files#r37242366 for my thoughts on this method.